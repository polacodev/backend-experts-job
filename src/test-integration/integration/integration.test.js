import { createTestClient } from 'apollo-server-testing';

import server from '../../index';

describe('Integration testing component', () => {
  const { mutate, query } = createTestClient(server);

  // TEST USERS
  describe('user component', () => {
    it('createUser mutation for first user', async () => {
      const CREATE_USER_1 = `
      mutation CreateUser($user: UserInput!) {
        createUser(user: $user) {
          name
          email
          password
          workarea
          status
          cellphone
        }
      }
    `;

      const response = await mutate({
        mutation: CREATE_USER_1,
        variables: {
          user: {
            name: 'ximena',
            email: 'ximena@gmail.com',
            password: 'sucmar123',
            cellphone: '67543214',
            workarea: 'developer',
            status: false,
            description: 'ximena description',
            knowledge: 'ximena knowledge',
          },
        },
      });
      expect(response).toMatchSnapshot();
    });

    it('createUser mutation for second user', async () => {
      const CREATE_USER_2 = `
      mutation CreateUser($user: UserInput!) {
        createUser(user: $user) {
          name
          email
          password
          workarea
          status
          cellphone
        }
      }
    `;

      const response = await mutate({
        mutation: CREATE_USER_2,
        variables: {
          user: {
            name: 'rolan',
            email: 'rolan@gmail.com',
            password: 'rolan123',
            cellphone: '67543214',
            workarea: 'developer',
            status: false,
            description: 'rolan description',
            knowledge: 'rolan knowledge',
          },
        },
      });
      expect(response).toMatchSnapshot();
    });

    it('getAllUsers query', async () => {
      const GET_ALL_USERS = `
      query GetUsers($search: SearchUserInput) {
        getUsers(search: $search) {
          name
          email
          password
          cellphone
          workarea
        }
      }
    `;

      const response = await query({
        query: GET_ALL_USERS,
        variables: {
          search: {
            name: {
              contains: 'la',
            },
          },
        },
      });
      expect(response).toMatchSnapshot();
    });

    it('getCustomUsers query using name field', async () => {
      const GET_ALL_USERS = `
      query GetUsers($search: SearchUserInput) {
        getUsers(search: $search) {
          name
          email
          password
          cellphone
          workarea
        }
      }
    `;

      const response = await query({
        query: GET_ALL_USERS,
        variables: {
          search: {
            workarea: {
              contains: 'dev',
            },
          },
        },
      });
      expect(response).toMatchSnapshot();
    });

    it('getCustomUsers query using status field', async () => {
      const GET_CUSTOM_FILTER = `
      query GetUsers($search: SearchUserInput) {
        getUsers(search: $search) {
          name
          email
          password
          cellphone
          workarea
        }
      }
    `;

      const response = await query({
        query: GET_CUSTOM_FILTER,
        variables: {
          search: {
            status: {
              eq: false,
            },
          },
        },
      });
      expect(response).toMatchSnapshot();
    });
  });

  // TEST AUTHENTICATION
  describe('authentication component', () => {
    it('get the user authentication', async () => {
      const GET_AUTHENTICATION = `
        query Authentication($auth: AuthInput!) {
          authentication(auth: $auth) {
            token
            isAuthenticated
          }
        }
      `;
      const response = await query({
        query: GET_AUTHENTICATION,
        variables: {
          auth: {
            email: 'marco.baltazar@gmail.com',
            password: 'elpepe',
          },
        },
      });
      expect(response).toMatchSnapshot();
    });
  });

  // NOTIFICATION TESTS
  // describe('Notification component', () => {
  // it('createNotification subscription for first notification', async () => {
  //   const CREATE_NOTIFICATION = `
  //     subscription UserAdded {
  //       userAdded {
  //         name
  //         email
  //         cellphone
  //         workarea
  //       }
  //     }
  //   `;

  //   const response = await query({ query: CREATE_NOTIFICATION });
  //   expect(response).toMatchSnapshot();
  // });

  // it('createNotification subscription for second notification', async () => {
  //   const CREATE_NOTIFICATION = `
  //     subscription createNotification {
  //       createNotification(notification: {
  //         user_id: "id_fake",
  //         message: "second notification"
  //       }) {
  //         user_id
  //         message
  //       }
  //     }
  //   `;

  //   const response = await query({ query: CREATE_NOTIFICATION });
  //   expect(response).toMatchSnapshot();
  // });

  //   it('getAllNotifications query', async () => {
  //     const GET_ALL_NOTIFICATIONS = `
  //       query GetNotifications {
  //         getNotifications {
  //           user_id
  //           createdBy
  //           message
  //           name
  //           email
  //         }
  //       }
  //     `;

  //     const response = await query({ query: GET_ALL_NOTIFICATIONS });
  //     expect(response).toMatchSnapshot();
  //   });

  //   it('getAllNotifications by user id query', async () => {
  //     const GET_ALL_NOTIFICATIONS_BY_USER_ID = `
  //       query GetNotificationsByUserId($id: ID!) {
  //         getNotificationsByUserId(_id: $id) {
  //           user_id
  //           createdBy
  //           message
  //           name
  //         }
  //       }
  //     `;

  //     const response = await query({
  //       query: GET_ALL_NOTIFICATIONS_BY_USER_ID,
  //       variables: {
  //         id: '61d820e106bbd50a38a40b05',
  //       },
  //     });
  //     expect(response).toMatchSnapshot();
  //   });
  // });

  // RATE COMPONENT
  describe('Rate component', () => {
    it('Get all Rates', async () => {
      const GET_ALL_RATES = `
        query GetRates {
          getRates {
            user_id
            ratedBy
            name
            rate
          }
        }
      `;
      const response = await query({ query: GET_ALL_RATES });
      expect(response).toMatchSnapshot();
    });

    it('Get Rate by User Id', async () => {
      const GET_RATE_BY_USER_ID = `
        query GetRatesByUserId($id: ID!) {
          getRatesByUserId(_id: $id) {
            _id
            user_id
            ratedBy
            name
            rate
          }
        }
      `;
      const response = await query({
        query: GET_RATE_BY_USER_ID,
        variables: {
          id: '61d820e106bbd50a38a40b05',
        },
      });
      expect(response).toMatchSnapshot();
    });

    it('Get Rate Average by User Id', async () => {
      const GET_RATE_AVERAGE_BY_USER_ID = `
        query GetRatesAverageByUserId($id: ID!) {
          getRatesAverageByUserId(_id: $id) {
            user_id
            averageRate
          }
        }
      `;
      const response = await query({
        query: GET_RATE_AVERAGE_BY_USER_ID,
        variables: {
          id: '61d820e106bbd50a38a40b05',
        },
      });
      expect(response).toMatchSnapshot();
    });
  });

  // CONTACT COMPONENT
  describe('Contacts component', () => {
    it('Get all Contacts', async () => {
      const GET_ALL_CONTACTS = `
          query GetContacts {
            getContacts {
              user_id
              createdBy
              name
              email
            }
          }
        `;
      const response = await query({ query: GET_ALL_CONTACTS });
      expect(response).toMatchSnapshot();
    });

    it('Get Contact by User Id', async () => {
      const GET_CONTACT_BY_USER_ID = `
        query GetContactsByUserId($id: ID!) {
          getContactsByUserId(_id: $id) {
            user_id
            createdBy
            name
            email
            cellphone
          }
        }
      `;
      const response = await query({
        query: GET_CONTACT_BY_USER_ID,
        variables: {
          id: '61d820e106bbd50a38a40b05',
        },
      });
      expect(response).toMatchSnapshot();
    });
  });
});
