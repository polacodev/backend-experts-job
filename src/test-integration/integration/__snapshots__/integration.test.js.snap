// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration testing component Contacts component Get Contact by User Id 1`] = `
Object {
  "data": Object {
    "getContactsByUserId": Array [],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration testing component Contacts component Get all Contacts 1`] = `
Object {
  "data": Object {
    "getContacts": Array [],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration testing component Rate component Get Rate Average by User Id 1`] = `
Object {
  "data": Object {
    "getRatesAverageByUserId": Object {
      "averageRate": 0,
      "user_id": "61d820e106bbd50a38a40b05",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration testing component Rate component Get Rate by User Id 1`] = `
Object {
  "data": Object {
    "getRatesByUserId": Array [],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration testing component Rate component Get all Rates 1`] = `
Object {
  "data": Object {
    "getRates": Array [],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration testing component authentication component get the user authentication 1`] = `
Object {
  "data": Object {
    "authentication": Object {
      "isAuthenticated": false,
      "token": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration testing component user component createUser mutation for first user 1`] = `
Object {
  "data": Object {
    "createUser": Object {
      "cellphone": "67543214",
      "email": "ximena@gmail.com",
      "name": "ximena",
      "password": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.InN1Y21hcjEyMyI.oY9heCKZNZmpjW-8iS1S4xS1A5C0oTeDsBjlaudrQDs",
      "status": false,
      "workarea": "developer",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration testing component user component createUser mutation for second user 1`] = `
Object {
  "data": Object {
    "createUser": Object {
      "cellphone": "67543214",
      "email": "rolan@gmail.com",
      "name": "rolan",
      "password": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.InJvbGFuMTIzIg.4VT_HgmMjXZjbaBsLEJnIWBg_WPmJGf9yT7idlL1rok",
      "status": false,
      "workarea": "developer",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration testing component user component getAllUsers query 1`] = `
Object {
  "data": Object {
    "getUsers": Array [
      Object {
        "cellphone": "67543214",
        "email": "rolan@gmail.com",
        "name": "rolan",
        "password": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.InJvbGFuMTIzIg.4VT_HgmMjXZjbaBsLEJnIWBg_WPmJGf9yT7idlL1rok",
        "workarea": "developer",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration testing component user component getCustomUsers query using name field 1`] = `
Object {
  "data": Object {
    "getUsers": Array [
      Object {
        "cellphone": "67543214",
        "email": "ximena@gmail.com",
        "name": "ximena",
        "password": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.InN1Y21hcjEyMyI.oY9heCKZNZmpjW-8iS1S4xS1A5C0oTeDsBjlaudrQDs",
        "workarea": "developer",
      },
      Object {
        "cellphone": "67543214",
        "email": "rolan@gmail.com",
        "name": "rolan",
        "password": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.InJvbGFuMTIzIg.4VT_HgmMjXZjbaBsLEJnIWBg_WPmJGf9yT7idlL1rok",
        "workarea": "developer",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Integration testing component user component getCustomUsers query using status field 1`] = `
Object {
  "data": Object {
    "getUsers": Array [
      Object {
        "cellphone": "67543214",
        "email": "ximena@gmail.com",
        "name": "ximena",
        "password": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.InN1Y21hcjEyMyI.oY9heCKZNZmpjW-8iS1S4xS1A5C0oTeDsBjlaudrQDs",
        "workarea": "developer",
      },
      Object {
        "cellphone": "67543214",
        "email": "rolan@gmail.com",
        "name": "rolan",
        "password": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.InJvbGFuMTIzIg.4VT_HgmMjXZjbaBsLEJnIWBg_WPmJGf9yT7idlL1rok",
        "workarea": "developer",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
